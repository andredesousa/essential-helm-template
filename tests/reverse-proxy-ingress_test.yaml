suite: Ingress test deployment
values:
  - ../values.yaml
templates:
  - reverse-proxy-ingress.yaml
tests:
  - it: should pass all kinds of assertion
    documentIndex: 0
    asserts:
      - hasDocuments:
          count: 2
      - isAPIVersion:
          of: networking.k8s.io/v1
      - isKind:
          of: Ingress
      - equal:
          path: metadata.annotations
          value:
            nginx.ingress.kubernetes.io/ssl-redirect: "false"
            nginx.ingress.kubernetes.io/use-regex: "true"
            nginx.ingress.kubernetes.io/rewrite-target: /$2
      - contains:
          path: spec.rules[0].http.paths
          content:
            path: /v1(/|$)(.*)
            pathType: Prefix
            backend:
              service:
                name: aks-current-api
                port:
                  number: 80
      - contains:
          path: spec.rules[0].http.paths
          content:
            path: /v2(/|$)(.*)
            pathType: Prefix
            backend:
              service:
                name: aks-future-api
                port:
                  number: 80
      - matchSnapshot:
          path: spec
