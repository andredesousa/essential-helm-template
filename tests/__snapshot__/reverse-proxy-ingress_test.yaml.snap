should pass all kinds of assertion:
  1: |
    ingressClassName: nginx
    rules:
    - http:
        paths:
        - backend:
            service:
              name: aks-current-api
              port:
                number: 80
          path: /v1(/|$)(.*)
          pathType: Prefix
        - backend:
            service:
              name: aks-future-api
              port:
                number: 80
          path: /v2(/|$)(.*)
          pathType: Prefix
        - backend:
            service:
              name: aks-current-api
              port:
                number: 80
          path: /(.*)
          pathType: Prefix
